src/lumigo_tracer/spans_container.py

```diff
@@ -150,44 +150,47 @@ def add_span(self, span: dict) -> dict:
         This function parses an request event and add it to the span.
         """
         new_span = recursive_json_join(span, self.base_msg)
-        self.spans.append(new_span)
-        self.span_ids_to_send.add(span["id"])
+        span_id = new_span["id"]
+        self.spans[span_id] = new_span
+        self.span_ids_to_send.add(span_id)
```

Comment:
Why do we need `span_ids_to_send`? Maybe we don't need it anymore?

Reply:
We're using it in order to not send spans twice, given the timeout mechanism:
It is relevant only to the case that the TO mechanism has been executed, and then the lambda *ended normally*
So we don't want to send the same spans twice :)